src/status_im/infra/market/events.cljs

```diff
@@ -0,0 +1,77 @@
+(ns status-im.infra.market.events
+  (:require
+    [status-im.infra.market.core :as core]
+    [status-im.infra.market.transform :as t]
+    [utils.number]
+    [utils.re-frame :as rf]))
+
+(def ^:const leaderboard-request-page-size 20)
+
+(rf/reg-event-fx :infra.market/start-fetching-tokens
+ (fn [{:keys [db]}]
+   (let [start-page 1]
+     {:db (assoc db :infra.market/last-fetched-page start-page)
+      :fx [[:dispatch [:infra.market/fetch-token-page start-page leaderboard-request-page-size]]]})))
+
+(rf/reg-event-fx :infra.market/fetch-more-tokens
+ (fn [{:keys [db]}]
+   (let [last-page               (:infra.market/last-fetched-page db)
+         next-page               (inc last-page)
+         should-fetch-next-page? (> last-page 0)]
+     (when should-fetch-next-page?
+       {:db (assoc db :infra.market/last-fetched-page next-page)
+        :fx [[:dispatch [:infra.market/fetch-token-page next-page leaderboard-request-page-size]]]}))))
+
+(rf/reg-event-fx :infra.market/ensure-tokens-updating
+ (fn [{:keys [_db]} [token-position tokens-amount]]
+   (let [{:keys [page-size page-index]} (core/page-size-to-ensure-tokens-updating
+                                         {:first-visible-index token-position
+                                          :visible-count       tokens-amount})]
+     {:fx [[:dispatch [:infra.market/fetch-token-page page-index page-size]]]})))
+
+(rf/reg-event-fx :infra.market/fetch-token-page
+ (fn [{:keys [_db]} [page page-size]]
+   (let [page       page
+         sort-order 0
+         currency   ""
+         params     [page
+                     page-size
+                     sort-order
+                     currency]]
+     {:fx [[:dispatch
+            [:infra.rpc/call
+             {:method "wallet_fetchMarketTokenPageAsync"
+              :params params}]]]})))
+
+(rf/reg-event-fx :infra.market/stop-fetching-tokens
+ (fn [{:keys [db]}]
+   {:db (assoc db :infra.market/last-fetched-page 0)
+    :fx [[:dispatch
+          [:infra.rpc/call
+           {:method "wallet_unsubscribeFromLeaderboard"
+            :params []}]]]}))
+
+(rf/reg-event-fx :infra.market/leaderboard-page-fetched
+ (fn [{:keys [_db]} [payload]]
+   (let [message (t/signal-payload-json->clj payload)]
+     {:fx [[:dispatch [:infra.market/tranform-and-save-page-data message]]]})))
+
+(rf/reg-event-fx :infra.market/leaderboard-page-data-updated
+ (fn [{:keys [_db]} [payload]]
+   (let [message (t/signal-payload-json->clj payload)]
+     {:fx [[:dispatch [:infra.market/tranform-and-save-page-data message]]]})))
+
+(rf/reg-event-fx :infra.market/tranform-and-save-page-data
```

Comment:
This is an example of how we can split/normalize data and store it under different rf-db keys instead of putting `status-go` structure almost "as-is"